src/servant/tt_002/3678.c

```diff
@@ -103,7 +103,52 @@ void CreateEventEntity_Dupe(Entity* entityParent, s32 entityId, s32 params) {
     }
 }
 
-INCLUDE_ASM("servant/tt_002/nonmatchings/3678", func_us_80173BD0);
+void func_us_80173BD0(Entity* arg0) {
+    if (abs(arg0->velocityY) > abs(arg0->velocityX)) {
+        if (abs(arg0->velocityY) < 0x8000) {
```

Comment:
All the velocity numbers can be FIXED

Reply:
Oh yeah. I’ll fix that tonight and update the PR